examples/cfd/external_aerodynamics/domino_sensitivity/main.py

```diff
@@ -0,0 +1,491 @@
+                    [0.98881036, 0.00550783, 0.00854675, 0.00452144],
+                    [-2.4203062, -0.00740275, -0.00848471, -0.00448634],
+                ],
+                dtype=np.float32,
+            )
+        ).to(self.device)
+
+    def __call__(
+        self,
+        mesh: pv.PolyData,
+        stream_velocity: float = 38.889,
+        stencil_size: int = 7,
+        air_density: float = 1.205,
+    ) -> dict[str, np.ndarray]:
+        """Performs DoMINO inference on a given geometry to predict aerodynamic quantities.
+
+        This method takes a PyVista mesh representing a 3D geometry and computes the
+        aerodynamic predictions using the DoMINO model. It handles the data preprocessing,
+        model inference, and post-processing of results.
+
+        Args:
+            mesh: PyVista PolyData mesh representing the 3D geometry to analyze
+            stream_velocity: Inlet flow velocity in m/s. Defaults to 38.889 m/s.
+            stencil_size: Number of neighboring points to consider for surface calculations.
+                Defaults to 7.
+            air_density: Air density in kg/m³. Defaults to 1.205 kg/m³.
+
+        Returns:
+            dict: Dictionary containing the following keys:
+                - 'geometry_coordinates': Array of geometry point coordinates
+                - 'geometry_normal_sensitivity': Array of sensitivity values for each point
+                - 'pred_surf_pressure': Array of predicted surface pressure values [Pa]
+                - 'pred_surf_wall_shear_stress': Array of predicted wall shear stress values [τx, τy, τz] [Pa]
+                - 'aerodynamic_force': Array of total computed aerodynamic force [Fx, Fy, Fz] [N]
+
+        Example:
+            >>> import pyvista as pv
+            >>> from domino_sensitivity import DoMINOInference
+            >>>
+            >>> # Load geometry
+            >>> mesh = pv.read("car.stl")
+            >>>
+            >>> # Initialize inference
+            >>> domino = DoMINOInference(cfg)
+            >>>
+            >>> # Run inference
+            >>> results = domino(
+            ...     mesh=mesh,
+            ...     stream_velocity=30.0,
+            ...     stencil_size=7,
+            ...     air_density=1.205
+            ... )
+            >>>
+            >>> # Access results
+            >>> forces = results['aerodynamic_force']
+            >>> print(f"Drag force: {forces[0]:.2f} N")
+        """
+
+        datapipe = DesignDatapipe(
+            mesh=mesh,
+            bounding_box=self.bounding_box_min_max,
+            bounding_box_surface=self.bounding_box_surface_min_max,
+            grid_resolution=self.cfg.model.interp_res,
+            stencil_size=stencil_size,
+        )
+        dataloader = torch.utils.data.DataLoader(
+            datapipe, batch_size=2**13, shuffle=False
+        )
+
+        input_dict = {
+            k: torch.from_numpy(np.expand_dims(np.float32(v), axis=0)).to(self.device)
```

Comment:
Nit: Consider doing the conversion to torch + transfer to the device in the datapipe.  Would enable numpy/cupy agnostic loading more easily.

Reply:
Good call, thanks! Fixed: f79d2698c9fe4ac8fd9123b3aac76422d7519f6a